src/test/merkle_tests.cpp

```diff
@@ -237,12 +238,12 @@ BOOST_AUTO_TEST_CASE(merkle_test_BlockWitness)
     uint256 blockWitness = BlockWitnessMerkleRoot(block);
 
     std::vector<uint256> hashes;
-    hashes.resize(block.vtx.size());
-    hashes[0].SetNull();
-    hashes[1] = block.vtx[1]->GetHash().ToUint256();
+    hashes.resize(vtx_count); // Note: leaving odd count to exercise old behavior
+    for(size_t pos{1}; pos < vtx_count; pos++) {
```

Comment:
style: space after `for` and `++i` is preferred over `i++` (from `doc/developer-notes.md`).
```suggestion
    for (size_t pos{1}; pos < vtx_count; ++pos) {
```
(the loop a few lines above also uses `pos++`)